assets/js/common/DateFilter/DateFilter.jsx

```diff
@@ -0,0 +1,213 @@
+import React, { Fragment, useState, useRef } from 'react';
+import classNames from 'classnames';
+import { Transition } from '@headlessui/react';
+
+import useOnClickOutside from '@hooks/useOnClickOutside';
+import { EOS_CLOSE } from 'eos-icons-react';
+
+
+const oneHour = 60 * 60 * 1000;
+const preconfiguredOptions = {
+  '1h ago': () => new Date(Date.now() - oneHour),
+  '24h ago': () => new Date(Date.now() - 24 * oneHour),
+  '7d ago': () => new Date(Date.now() - 7 * 24 * oneHour),
+  '30d ago': () => new Date(Date.now() - 30 * 24 * oneHour),
+};
+
+const renderOptionItem = (option, placeholder) => {
+  if (!option || !Array.isArray(option)) {
+    return placeholder;
+  }
+  if (typeof option[2] === 'function') {
+    return option[2]();
+  }
+  return option[0];
+};
+
+
+const parseInputOptions = (options) =>
+  options
+    .map((option) => {
+      if (typeof option === 'string' && option in preconfiguredOptions) {
+        return [option, preconfiguredOptions[option]];
+      }
+      if (
+        Array.isArray(option) &&
+        typeof option[1] === 'function' &&
+        option[1]() instanceof Date
```

Comment:
**decision** Options come with a value function so that the actual reference date is calculated at selection time. That's simpler for the consumer as it already has a date, as expected from a date filter, which can just store the value and use it at its convenience.

That comes with a little drawback: if the date reflects a period (for example, 10 minutes ago), that value is never updated unless a new selection is made.

Another viable solution would be to return a number instead of a Date object; such a number can represent the milliseconds back from now. Being a relative value it does not have to update. However, the using component must handle the calculation itself.